src/web_fetch.js

```diff
@@ -158,39 +174,35 @@ async function webFetch(url, options) {
 
       if (multipart) {
         mimeType = 'multipart/byteranges';
-      } else if (mime.getType(nodePath.extname(filePath))) {
-        mimeType = mime.getType(nodePath.extname(filePath));
-      } else {
-        mimeType = 'application/octet-stream';
       }
 
-      response = {
+      const response = {
         headers: {
           'Content-Type': mimeType
         },
         body: data
       };
 
       if (range && multipart) {
-        response = {
-          headers: {
-            'Content-Type': mimeType,
-            'Content-Length': JSON.stringify(data).length
-          },
-          parts: data
-        };
+        response.headers['Content-Length'] = JSON.stringify(data).length;
+        response.parts = data;
```

Comment:
Minor: If you place a log somewhere after this point you'll see both the `parts` and `body` properties in the `data` object.
```
{ headers:
   { 'Content-Type': 'multipart/byteranges',
     'Content-Length': 365 },
  body:
   [ { body: <Buffer@0x0000026677A49E10 6c 6c 6f>,
       headers: [Object] },
     { body: <Buffer@0x0000026677A49E18 6f 72 6c>,
       headers: [Object] },
     { body: <Buffer@0x0000026677A49E20 6c 6f 72 65 6d>,
       headers: [Object] } ],
  parts:
   [ { body: <Buffer@0x0000026677A49E10 6c 6c 6f>,
       headers: [Object] },
     { body: <Buffer@0x0000026677A49E18 6f 72 6c>,
       headers: [Object] },
     { body: <Buffer@0x0000026677A49E20 6c 6f 72 65 6d>,
       headers: [Object] } ] }
```
Should `body` be a null value or just `delete`d altogether?

Reply:
Nice catch @hunterlester , I think it should be `delete`ed, I will try to also add some tests which make sure the attributes returned are exactly those expected and no more for each case.